### PR TITLE
Add client for DCOS_OSS-5101

### PR DESCRIPTION
**Pull request by br-lewis**

### CODE REVIEW

File: api/rest/client.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	Status(ctx context.Context, node string, id string) (*Bundle, error)
+	// GetFile downloads bundle file of bundle with given id from node at the given
+	// url and save it to local filesystem. It returns path where file is stored or
+	// error if there were a problem.
+	GetFile(ctx context.Context, node string, id string) (path string, err error)
+}
+
+type BundleStatus struct {
+	ID   string
+	node node
```
Comment (janisz) [inline api/rest/client.go]: We need node definition here

File: api/rest/client.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+// Client is an interface that can talk with dcos-diagnostics REST API and manipulate remote bundles
+type Client interface {
+	// Create ask node url to start bundle creation process with given id
+	Create(ctx context.Context, node string, id string) (*Bundle, error)
+	// Status returns status of bundle with given id on node at the given url
+	Status(ctx context.Context, node string, id string) (*Bundle, error)
+	// GetFile downloads bundle file of bundle with given id from node at the given
+	// url and save it to local filesystem. It returns path where file is stored or
+	// error if there were a problem.
+	GetFile(ctx context.Context, node string, id string) (path string, err error)
```
Comment (janisz) [inline api/rest/client.go]:
```suggestion
	GetFile(ctx context.Context, node string, id string, output string) err error
```

File: api/rest/client.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	url := fmt.Sprintf("%s/system/health/v1/diagnostics/%s", node, id)
+
+	logrus.WithField("bundle", id).WithField("url", url).Info("sending bundle creation request")
+
+	type payload struct {
+		Type string `json:"type"`
+	}
+
+	body, err := json.Marshal(payload{
+		Type: "LOCAL",
```
Comment (janisz) [inline api/rest/client.go]: Move it to constant

File: api/rest/client.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Client is an interface that can talk with dcos-diagnostics REST API and manipulate remote bundles
+type Client interface {
+	// Create ask node url to start bundle creation process with given id
```
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	// Create requests the given node to start a bundle creation process with that is identified by the given ID
```

File: api/rest/client.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	Create(ctx context.Context, node string, id string) (*Bundle, error)
```
Comment (makkes) [inline api/rest/client.go]: Is it possible to make `node` be of type `net/url.URL`?

Comment (makkes) [inline api/rest/client.go]:
```suggestion
	Create(ctx context.Context, node string, ID string) (*Bundle, error)
```
https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: api/rest/client.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	// Status returns status of bundle with given id on node at the given url
```
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	// Status returns the status of the bundle with the given ID on the given node
```

File: api/rest/client.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+	Status(ctx context.Context, node string, id string) (*Bundle, error)
```
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	Status(ctx context.Context, node string, ID string) (*Bundle, error)
```

File: api/rest/client.go [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+	// GetFile downloads bundle file of bundle with given id from node at the given
```
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	// GetFile downloads the bundle file of the bundle with the given ID from the node
```

File: api/rest/client.go [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+	// url and save it to local filesystem. It returns path where file is stored or
```
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	// and saves it to the local filesystem. It returns the path where the file is stored or
```

File: api/rest/client.go [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+	// error if there were a problem.
```
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	// error if there was a problem.
```

File: api/rest/client.go (hunk 2 above)
Comment (makkes) [inline api/rest/client.go]:
```suggestion
	GetFile(ctx context.Context, node string, ID string) (path string, err error)
```

File: api/rest/client.go (hunk 5 above)
Comment (makkes) [inline api/rest/client.go]: I'd prefer to name this method `CreateBundle` to make it clear what this is about. It's also more in line with the `GetFile` method's name.

File: api/rest/client.go [hunk 11]
```diff
@@ -62,16 +63,21 @@ func (d DiagnosticsClient) Create(ctx context.Context, node string, id string) (
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("unexpected HTTP status code received: %d", resp.StatusCode)
+		body, e := ioutil.ReadAll(resp.Body)
```
Comment (makkes) [inline api/rest/client.go]: Umm, no. We're reading the full body here just to cut off everything beyond the first 100 characters. Why not read only the first 100 bytes to begin with?

File: api/rest/client.go (hunk 5 above)
Comment (janisz) [reply api/rest/client.go]: Good idea

File: api/rest/client.go (hunk 11 above)
Comment (janisz) [reply api/rest/client.go]: Thanks, fixed

File: api/rest/client.go [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+	"github.com/sirupsen/logrus"
+)
+
+const bundlesEndpoint = "/system/health/v1/diagnostics"
+
+// Client is an interface that can talk with dcos-diagnostics REST API and manipulate remote bundles
+type Client interface {
+	// CreateBundle requests the given node to start a bundle creation process with that is identified by the given ID
+	CreateBundle(ctx context.Context, node string, ID string) (*Bundle, error)
```
Comment (janisz) [inline api/rest/client.go]: I haven't spot it before. Why do we have pointer here? It's a small struct so IMO it should be a value

**Pull request merged**